test-framework/src/main/java/org/keycloak/test/framework/page/WelcomePage.java

```diff
@@ -45,4 +55,17 @@ public void assertUserCreated() {
         Assertions.assertTrue(pageAlert.getText().contains("User created"));
     }
 
+    public boolean isPasswordSet() {
```

Comment:
This seems like it could use a more specific selector rather than querying the whole page.

Reply:
The HTML structure differs a little bit for local and remote access. It probably can be done but this way its simpler and more maintainable imo. 

Considering its used by just 1 test class do we need to consider performance and optimize this ?

Reply:
What I meant is difference like this 

local
![image](https://github.com/user-attachments/assets/d9718a1d-acbb-4ecf-8ad7-3238ef45ab5c)

public
![image](https://github.com/user-attachments/assets/11e14511-d4b3-4d28-8734-0ed94bd0c9fc)

The message is overall same however the structure seems different. I guess it could have 2 different asserts for local and public setting in order to be more specific ?

Reply:
I see you managed to resolve this so I will close this thread